### PR TITLE
Remove get process and software name functions from versions.yml creation

### LINKED ISSUE #1284: Use Jinja2 to get process/module name in main script of module template
The idea would be to remove the need for any custom functions to get the process and module name here:
https://github.com/nf-core/tools/blob/7ca734bfbf1da3fac6c1e729590149877259c397/nf_core/module-template/modules/main.nf#L81-L84

We would replace these lines with some Jinja Ninja replacement. That way the correct fields will be verbosely populated there and we won't have to invoke a function.

### PR DESCRIPTION
**Pull request by JoseEspinosa**
Remove custom functions `getProcessName` and `getSoftwareName` and instead use the template to directly populate the process name and the software name within the `versions.yml` code.

Closes #1284

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/module-template/modules/main.nf [hunk 1]
```diff
@@ -79,8 +79,8 @@ process {{ tool_name_underscore|upper }} {
         $bam
 
     cat <<-END_VERSIONS > versions.yml
-    ${getProcessName(task.process)}:
-        ${getSoftwareName(task.process)}: \$( samtools --version 2>&1 | sed 's/^.*samtools //; s/Using.*\$//' )
+    {{ tool_name_underscore|upper }}:
```
Comment (grst) [inline nf_core/module-template/modules/main.nf:82]: Didn't we go for `${task.process}` here?

**Review by grst — COMMENTED** [review]
There are a few other things that need to be updated in the module pipeline, e.g
 * get rid of functions.nf
 * get rid of publish dir
 
but is that supposed to be part of this PR?

File: nf_core/module-template/modules/main.nf (hunk 1 above)
Comment (JoseEspinosa) [reply nf_core/module-template/modules/main.nf]: Yeah, forgot to push the change.

**Comment by JoseEspinosa** [comment]
> There are a few other things that need to be updated in the module pipeline, e.g
> 
>     * get rid of functions.nf
> 
>     * get rid of publish dir
> 
> 
> but is that supposed to be part of this PR?

I am addressing all the other changes in this other PR, at the beginning I was keeping separate the update of the modules template and the pipeline template, but they have some dependencies, so I went for a single draft here: https://github.com/nf-core/tools/pull/1327 
I will add the things we have just discussed and ping you there.

**Comment by JoseEspinosa** [comment]
Thinking again, might be worth to close this draft and add everything in the same PR? 🤔

**Comment by grst** [comment]
Yeah, let's do that :rocket:

**Comment by JoseEspinosa** [comment]
These changes are addressed in #1327 and thus, this draft PR has been closed

**Pull request closed without merging**